controllers/datadogagent/feature/apm/feature.go

```diff
@@ -228,6 +266,48 @@ func (f *apmFeature) ManageDependencies(managers feature.ResourceManagers, compo
 // ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
 // It should do nothing if the feature doesn't need to configure it.
 func (f *apmFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	if f.singleStepInstrumentation != nil {
+		if len(f.singleStepInstrumentation.disabledNamespaces) > 0 && len(f.singleStepInstrumentation.enabledNamespaces) > 0 {
+			// This configuration is not supported
+			return fmt.Errorf("`enabledNamespaces` and `disabledNamespaces` cannot be set together")
```

Comment:
Behavior I saw when testing https://github.com/DataDog/helm-charts/pull/1252 was not to error and instead use the list based on the flag selection - i.e. if enabled use `disabledNamespaces`, if disabled use `enabledNamespaces`. Returning error will prevent reconciling and any agent installation.

Could you confirm this is what we want, @liliyadd?